File: app/src/main/java/ru/binaryblitz/sportup/server/DeviceInfoStore.java
package ru.binaryblitz.sportup.server;

import android.content.Context;
import android.content.SharedPreferences;
import ru.binaryblitz.Chisto.Server.ServerConfig;
import ru.binaryblitz.sportup.models.City;
import ru.binaryblitz.sportup.models.User;
import ru.binaryblitz.sportup.utils.LogUtil;

public class DeviceInfoStore {
    public static void saveToken(Context context, String token) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        prefs.edit().putString(ServerConfig.INSTANCE.getTokenEntity(), token).apply();
    }

    public static void resetToken(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        prefs.edit().putString(ServerConfig.INSTANCE.getTokenEntity(), "null").apply();
    }

    public static String getToken(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        return prefs.getString(ServerConfig.INSTANCE.getTokenEntity(), "null");
    }

    public static City getCityObject(Context context) {
        try {
            return City.Companion.fromString(DeviceInfoStore.getCity(context));
        } catch (Exception e) {
            return null;
        }
    }

    public static void saveUser(Context context, User user) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        prefs.edit().putString(ServerConfig.INSTANCE.getUserEntity(), user.asString()).apply();
    }

    public static User getUserObject(Context context) {
        try {
            return User.Companion.fromString(DeviceInfoStore.getUser(context));
        } catch (Exception e) {
            LogUtil.logException(e);
            return null;
        }
    }

    public static String getUser(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        return prefs.getString(ServerConfig.INSTANCE.getUserEntity(), "null");
    }

    public static void resetUser(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        prefs.edit().putString(ServerConfig.INSTANCE.getUserEntity(), "null").apply();
    }

    public static void saveCity(Context context, City city) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        prefs.edit().putString(ServerConfig.INSTANCE.getCityEntity(), city.asString()).apply();
    }


    public static String getCity(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        return prefs.getString(ServerConfig.INSTANCE.getCityEntity(), "null");
    }

    public static void resetCity(Context context) {
        SharedPreferences prefs = context.getSharedPreferences(
                ServerConfig.INSTANCE.getPrefsName(), Context.MODE_PRIVATE);
        prefs.edit().putString(ServerConfig.INSTANCE.getCityEntity(), "null").apply();
    }
}

File: app/src/main/java/ru/binaryblitz/SportUp/server/DeviceInfoStore.kt
package ru.binaryblitz.SportUp.server

import android.content.Context
import android.content.SharedPreferences
import ru.binaryblitz.SportUp.models.City
import ru.binaryblitz.SportUp.models.User
import ru.binaryblitz.SportUp.utils.LogUtil

object DeviceInfoStore {
    fun saveToken(context: Context, token: String) {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        prefs.edit().putString(ServerConfig.tokenEntity, token).apply()
    }

    fun resetToken(context: Context) {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        prefs.edit().putString(ServerConfig.tokenEntity, "null").apply()
    }

    fun getToken(context: Context): String {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        return prefs.getString(ServerConfig.tokenEntity, "null")
    }

    fun getCityObject(context: Context): City? {
        try {
            return City.fromString(DeviceInfoStore.getCity(context))
        } catch (e: Exception) {
            return null
        }
    }

    fun saveUser(context: Context, user: User) {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        prefs.edit().putString(ServerConfig.userEntity, user.asString()).apply()
    }

    fun getUserObject(context: Context): User? {
        try {
            return User.fromString(DeviceInfoStore.getUser(context))
        } catch (e: Exception) {
            LogUtil.logException(e)
            return null
        }
    }

    fun getUser(context: Context): String {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        return prefs.getString(ServerConfig.userEntity, "null")
    }

    fun resetUser(context: Context) {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        prefs.edit().putString(ServerConfig.userEntity, "null").apply()
    }

    fun saveCity(context: Context, city: City) {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        prefs.edit().putString(ServerConfig.cityEntity, city.asString()).apply()
    }


    fun getCity(context: Context): String {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        return prefs.getString(ServerConfig.cityEntity, "null")
    }

    fun resetCity(context: Context) {
        val prefs = context.getSharedPreferences(
                ServerConfig.preferencesName, Context.MODE_PRIVATE)
        prefs.edit().putString(ServerConfig.cityEntity, "null").apply()
    }
}
